syntax = "proto3";
package clamav;

service ClamAVScanner {
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
  rpc ScanFile(ScanFileRequest) returns (ScanResponse);
  rpc ScanStream(stream ScanStreamRequest) returns (ScanResponse);
  rpc ScanMultiple(stream ScanStreamRequest) returns (stream ScanResponse);
}

message HealthCheckRequest {}

message HealthCheckResponse {
  string status = 1;
  string message = 2;
}

message ScanFileRequest {
  bytes data = 1;
  string filename = 2;
}

message ScanStreamRequest {
  bytes chunk = 1;
  string filename = 2;
  bool is_last = 3;
}

message ScanResponse {
  string status = 1;
  string message = 2;
  double scan_time = 3;
  string filename = 4;
}
